### PR TITLE
Add retry workaround for error 502 to zypper_call

### PR DESCRIPTION
**Pull request by kravciak**
Local runs: (by @dzedro)
 - http://10.100.12.155/tests/6266#step/deepsea_testsuite/81 - acceptable failure
 - http://10.100.12.155/tests/6266#step/deepsea_testsuite/83 - fatal failure
 - http://10.100.12.155/tests/6291#step/deepsea_testsuite/5 - simulated retry with error 3

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -267,33 +267,26 @@ sub zypper_call {
...
-    else {
-        script_run("zypper -n $command; echo $str-\$?-$redirect", 0);
+    my $printer = $log ? "| tee /tmp/$log" : $dumb_term ? '| cat' : '';
+
+    # Retrying workarounds
+    my $ret;
+    for (1 .. 3) {
+        $ret = script_run("zypper -n $command $printer; ( exit \${PIPESTATUS[0]} )", $timeout);
+        if ($ret == 502) {
+            record_soft_failure "Retrying because of error 502";
```
Comment (coolo) [inline lib/utils.pm]: please no soft failure without bug reference. If there is no product bug to solve, the soft fail is worthless

File: lib/utils.pm [hunk 2]
```diff
@@ -267,33 +267,26 @@ sub zypper_call {
...
 
-    if ($ret) {
-        my ($ret_code) = $ret =~ /$str-(\d+)/;
-        unless (grep { $_ == $ret_code } @$allow_exit_codes) {
-            upload_logs('/var/log/zypper.log');
-            die "'zypper -n $command' failed with code $ret_code";
-        }
-
-        return $ret_code;
+    unless ($ret == any(@$allow_exit_codes)) {
```
Comment (coolo) [inline lib/utils.pm]: I don't think any works this way

Comment (kravciak) [reply lib/utils.pm]: fixed, it did not work - @dzedro keeps confusing me today..

**Comment by coolo** [comment]
I'm actually quite sure this will fail somehow - but I can't put the finger into it. So let's have it explode in production :)

**Comment by dzedro** [comment]
@coolo I have same feeling, but maybe codejack @kravciak knows what is he doing ^^

**Comment by kravciak** [comment]
img

**Comment by kravciak** [comment]
But I have the same feeling :)

**Comment by dzedro** [comment]
> But I have the same feeling :)

I had hope but you just destroyed it :cry:

**Comment by coolo** [comment]
I guess it was a bit naive to assume that zypper's exit status would be the http status code :)

 https://openqa.suse.de/tests/1527377#step/patch_and_reboot/133

**Comment by kravciak** [comment]
Finally we know where was `the feeling` coming from :) I'm testing fix in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4578

**Pull request merged**